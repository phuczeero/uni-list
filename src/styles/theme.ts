const fonts = {
  body:
    '-apple-system, BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji"',
};
const fontSizes = [12, 14, 16, 20, 24, 32, 48, 64, 72];
const space = [0, 4, 8, 16, 32, 64, 128, 256, 512];
const radii = {
  default: 6,
  circle: 99999,
};
const colors = {
  text: '#212529',
  muted: '#dee2e6',
  primary: '#007bff',
  danger: '#dc3545',
  blues: {
    light: '#dcedff',
    dark: '#004896',
  },
};
const sizes = {
  navbar: 64,
  spinner: 32,
};
const shadows = {
  card: '0 0 4px rgba(0, 0, 0, .125)',
};
const breakpoints = ['576px', '768px', '992px', '1200px'];

export const theme = {
  fonts,
  fontSizes,
  space,
  radii,
  colors,
  sizes,
  shadows,
  breakpoints,
};

export type ThemeType = typeof theme;
