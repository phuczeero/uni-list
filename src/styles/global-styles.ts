import { createGlobalStyle } from 'styled-components';
import { ThemeType } from './theme';

export const GlobalStyle = createGlobalStyle<{ theme: ThemeType }>`
  * {
    font-family: ${p => p.theme.fonts.body};
    color: ${p => p.theme.colors.text}
  }
`;
