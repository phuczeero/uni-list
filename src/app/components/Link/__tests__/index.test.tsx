import * as React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { theme } from 'styles/theme';
import { Link } from '..';

const renderWithTheme = () =>
  render(
    <MemoryRouter>
      <Link to="/" theme={theme}>
        Test
      </Link>
    </MemoryRouter>,
  );

describe('<Link  />', () => {
  it('should match snapshot', () => {
    const component = renderWithTheme();
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
