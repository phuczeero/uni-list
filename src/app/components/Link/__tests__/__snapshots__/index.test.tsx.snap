// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link  /> should match snapshot 1`] = `
.c0 {
  color: #007bff;
}

<a
  class="c0"
  href="/"
  theme="[object Object]"
>
  Test
</a>
`;
